basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/authentication/ElytronAuthenticationConfigurationTestCase.java

```diff
@@ -0,0 +1,528 @@
+     * Validate created resource is present in model.
+     */
+    @Test
+    public void testAddAuthenticationConfigurationTestCase() throws Exception {
+        final Address authenticationConfigurationAddress = elyOps.getElytronAddress(AUTHENTICATION_CONFIGURATION, RandomStringUtils.randomAlphanumeric(7));
+
+        page.navigate();
+
+        try {
+            page.switchToAuthenticationConfiguration()
+                    .getResourceManager()
+                    .addResource(AddAuthenticationConfigurationWizard.class)
+                    .name(authenticationConfigurationAddress.getLastPairValue())
+                    .saveAndDismissReloadRequiredWindowWithState()
+                    .assertWindowClosed();
+
+            Assert.assertTrue(page.getResourceManager().isResourcePresent(authenticationConfigurationAddress.getLastPairValue()));
+
+            new ResourceVerifier(authenticationConfigurationAddress, client).verifyExists();
+        } finally {
+            ops.removeIfExists(authenticationConfigurationAddress);
+            adminOps.reloadIfRequired();
+        }
+    }
+
+    /**
+     * @tpTestDetails Create Elytron Authentication configuration instance in model and try to remove it in Web Console's
+     * Elytron subsystem configuration.
+     * Validate the resource is not any more visible in Authentication configuration table.
+     * Validate created resource is not any more present in the model.
+     */
+    @Test
+    public void testRemoveAuthenticationConfigurationTestCase() throws Exception {
+        final Address authenticationConfigurationAddress = createAuthenticationConfiguration();
+
+        page.navigate();
+
+        try {
+            page.switchToAuthenticationConfiguration()
+                    .getResourceManager()
+                    .removeResource(authenticationConfigurationAddress.getLastPairValue())
+                    .confirmAndDismissReloadRequiredMessage()
+                    .assertClosed();
+
+            Assert.assertFalse(page.getResourceManager().isResourcePresent(authenticationConfigurationAddress.getLastPairValue()));
+
+            new ResourceVerifier(authenticationConfigurationAddress, client).verifyDoesNotExist();
+        } finally {
+            ops.removeIfExists(authenticationConfigurationAddress);
+            adminOps.reloadIfRequired();
+        }
+    }
+
+    /**
+     * @tpTestDetails Create Elytron Authentication configuration instance in model and try to edit its
+     * allow-all-mechanisms attribute value in Web Console's Elytron subsystem configuration.
+     * Validate edited attribute value in the model.
+     */
+    @Test
+    public void toggleAllowAllMechanisms() throws Exception {
+        final SnapshotBackup snapshotBackup = new SnapshotBackup();
+        client.apply(snapshotBackup.backup());
+
+        final Address authenticationConfigurationAddress = createAuthenticationConfiguration();
```

Comment:
Because if it fails, there is no resource added - thus nothing to remove in `finally`